### PR TITLE
EZP-30938: Removed usage of deprecated setCurrentUser method

### PR DESCRIPTION
**Pull request by ViniTou**
> Jira: https://jira.ez.no/browse/EZP-30938

### CODE REVIEW

File: lib/Behat/Context/ContentTypeContext.php [hunk 1]
```diff
@@ -44,7 +44,7 @@ final class ContentTypeContext extends RawMinkContext implements Context, Snippe
     public function __construct(Repository $repository, ContentTypeService $contentTypeService)
     {
         $this->repository = $repository;
-        $this->repository->setCurrentUser(new UserReference($this->adminUserId));
+        $this->repository->getPermissionResolver()->setCurrentUserReference(new UserReference($this->adminUserId));
```
Comment (alongosz) [inline lib/Behat/Context/ContentTypeContext.php]: Is it possible to inject PermissionResolver directly?

**Comment by ViniTou** [comment]
@alongosz 
done, but I have this feeling that this must be repeated for every suits that is using this conext, I am right @mnocon

File: behat_suites.yml [hunk 2]
```diff
@@ -7,21 +7,29 @@ repository-forms:
                 - vendor/ezsystems/repository-forms/features/ContentEdit
             contexts:
                 - EzSystems\RepositoryForms\Behat\Context\ContentEditContext
-                - EzSystems\RepositoryForms\Behat\Context\ContentTypeContext
+                - EzSystems\RepositoryForms\Behat\Context\ContentTypeContext:
+                    permissionResolver: '@eZ\Publish\API\Repository\PermissionResolver'
```
Comment (alongosz) [inline behat_suites.yml]: This wasn't here before, even though Context already had arguments, so I didn't expect it's needed.

Comment (ViniTou) [reply behat_suites.yml]: I was rather thinking about things like https://github.com/ezsystems/ezplatform-admin-ui/blob/master/behat_suites.yml#L36 as when removed dependency by annotations this is no longer built properly.

Comment (ViniTou) [reply behat_suites.yml]: as for one line change, injection with annotations does not work correctly (or I just couldn't find a way with service names using FCQN).

Comment (mnocon) [reply behat_suites.yml]: @ViniTou it's fixed by https://github.com/ezsystems/ezpublish-kernel/pull/2750

If you can wait a bit so I can merge it then using that would be the preffered approach, as keeping Context definitions in multiple behat_suites file is a tiresome thing.

**Comment by ViniTou** [comment]
This will be solved using @injectService when https://github.com/ezsystems/ezpublish-kernel/pull/2750 is merged :)

File: behat_suites.yml [hunk 3]
```diff
@@ -14,7 +14,7 @@ repository-forms:
             paths:
                 - vendor/ezsystems/repository-forms/features/FieldTypeForm
             contexts:
-                - EzSystems\RepositoryForms\Behat\Context\ContentTypeContext
+                - EzSystems\RepositoryForms\Behat\Context\ContentTypeContext:
```
Comment (mnocon) [inline behat_suites.yml]: Unneeded change 😉

Comment (ViniTou) [inline behat_suites.yml]:
```suggestion
                - EzSystems\RepositoryForms\Behat\Context\ContentTypeContext
```

Comment (ViniTou) [reply behat_suites.yml]: fixd

**Comment by alongosz** [comment]
Behat-related changes only (and passing), no further QA required,

**Pull request merged**